### ISSUE TITLE
Runtime failure: AttributeError: 'ColaQApplication' object has no attribute 'desktop'

### ISSUE THREAD

**Issue by fxcoudert** [git-cola/git-cola#1323]
When git-cola is built against Qt 6 as part of Homebrew, it installs fine but fails at runtime with:

```
$ git-cola                  
Traceback (most recent call last):
  File "/opt/homebrew/bin/git-cola", line 8, in <module>
    sys.exit(main())
             ^^^^^^
  File "/opt/homebrew/Cellar/git-cola/4.2.1/libexec/lib/python3.11/site-packages/cola/main.py", line 28, in main
    return args.func(args)
           ^^^^^^^^^^^^^^^
  File "/opt/homebrew/Cellar/git-cola/4.2.1/libexec/lib/python3.11/site-packages/cola/main.py", line 468, in cmd_cola
    view = MainView(context)
           ^^^^^^^^^^^^^^^^^
  File "/opt/homebrew/Cellar/git-cola/4.2.1/libexec/lib/python3.11/site-packages/cola/widgets/main.py", line 914, in __init__
    self.init_state(context.settings, self.set_initial_size)
  File "/opt/homebrew/Cellar/git-cola/4.2.1/libexec/lib/python3.11/site-packages/cola/widgets/standard.py", line 161, in init_state
    super(MainWindowMixin, self).init_state(settings, callback, *args, **kwargs)
  File "/opt/homebrew/Cellar/git-cola/4.2.1/libexec/lib/python3.11/site-packages/cola/widgets/standard.py", line 139, in init_state
    callback(*args, **kwargs)
  File "/opt/homebrew/Cellar/git-cola/4.2.1/libexec/lib/python3.11/site-packages/cola/widgets/main.py", line 948, in set_initial_size
    width, height = qtutils.desktop_size()
                    ^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/homebrew/Cellar/git-cola/4.2.1/libexec/lib/python3.11/site-packages/cola/qtutils.py", line 629, in desktop_size
    desk = desktop()
           ^^^^^^^^^
  File "/opt/homebrew/Cellar/git-cola/4.2.1/libexec/lib/python3.11/site-packages/cola/qtutils.py", line 625, in desktop
    return app().desktop()
           ^^^^^^^^^^^^^
AttributeError: 'ColaQApplication' object has no attribute 'desktop'
```

**Comment by fxcoudert** [comment]
I suppose this from the Qt 6 docs is relevant:

> QDesktopWidget and QApplication::desktop()
> 
> QDesktopWidget was already deprecated in Qt 5, and has been removed in Qt 6, together with QApplication::desktop().
> 
> [QScreen](https://doc.qt.io/qt-6/qscreen.html) provides equivalent functionality to query for information about available screens, screen that form a virtual desktop, and screen geometries.
> 
> Use [QWidget::setScreen](https://doc.qt.io/qt-6/qwidget.html#setScreen)() to create a [QWidget](https://doc.qt.io/qt-6/qwidget.html) on a specific display; note that this does not move a widget to a screen in a virtual desktop setup.

https://doc.qt.io/qt-6/widgets-changes-qt6.html#qdesktopwidget-and-qapplication-desktop

**Comment by davvid** [comment]
I believe this is already fixed on the main branch so it'll be fixed in the upcoming v4.3.0 release.
